utils/global-state-update-gen/src/generic/testing.rs

```diff
@@ -0,0 +1,562 @@
+    let result = get_update(&mut reader, config);
+
+    assert!(result.contains_key(&reader.get_seigniorage_recipients_key()));
+    assert_eq!(
+        result.get(&reader.get_total_supply_key()),
+        Some(&StoredValue::from(
+            CLValue::from_t(U512::from(305)).expect("should convert U512 to CLValue")
+        ))
+    );
+
+    // check purse write for validator1
+    let bid_purse = *reader
+        .get_bids()
+        .get(&validator1)
+        .expect("should have bid")
+        .bonding_purse();
+
+    assert_eq!(
+        result.get(&Key::Balance(bid_purse.addr())),
+        Some(&StoredValue::from(
+            CLValue::from_t(U512::zero()).expect("should convert U512 to CLValue")
+        ))
+    );
+
+    // check bid overwrite
+    let account1 = validator1.to_account_hash();
+    let mut expected_bid_1 = Bid::unlocked(validator1, bid_purse, U512::zero(), Default::default());
+    expected_bid_1.deactivate();
+    assert_eq!(
+        result.get(&Key::Bid(account1)),
+        Some(&StoredValue::from(expected_bid_1))
+    );
+
+    // check writes for validator2
+    let account2 = validator2.to_account_hash();
+
+    // the new account should be created
+    let account_write = result
+        .get(&Key::Account(account2))
+        .expect("should create account")
+        .as_account()
+        .expect("should be account")
+        .clone();
+    let main_purse_2 = account_write.main_purse();
+
+    // check that the main purse for the new account has been created with the correct amount
+    assert_eq!(
+        result.get(&Key::URef(main_purse_2)),
+        Some(&StoredValue::from(
+            CLValue::from_t(()).expect("should convert unit to CLValue")
+        ))
+    );
+    assert_eq!(
+        result.get(&Key::Balance(main_purse_2.addr())),
+        Some(&StoredValue::from(
+            CLValue::from_t(U512::from(102)).expect("should convert U512 to CLValue")
+        ))
+    );
+
+    let bid_write = result
+        .get(&Key::Bid(account2))
+        .expect("should create bid")
+        .as_bid()
+        .expect("should be bid")
+        .clone();
+    assert_eq!(bid_write.validator_public_key(), &validator2);
+    assert_eq!(
+        bid_write
+            .total_staked_amount()
+            .expect("should read total staked amount"),
+        U512::from(102)
+    );
+    assert!(!bid_write.inactive());
+
+    let bid_purse_2 = *bid_write.bonding_purse();
+
+    // check that the bid purse for the new validator has been created with the correct amount
+    assert_eq!(
+        result.get(&Key::URef(bid_purse_2)),
+        Some(&StoredValue::from(
+            CLValue::from_t(()).expect("should convert unit to CLValue")
+        ))
+    );
+    assert_eq!(
+        result.get(&Key::Balance(bid_purse_2.addr())),
+        Some(&StoredValue::from(
+            CLValue::from_t(U512::from(102)).expect("should convert U512 to CLValue")
+        ))
+    );
+
+    // 8 keys above should be all that was overwritten
```

Comment:
```suggestion
    // 10 keys above should be all that was overwritten
```

Reply:
Fixed in [8c25879](https://github.com/casper-network/casper-node/pull/3195/commits/8c25879b795846909ecad82d3f0d2b28ed137eb8)